lib/hexo/post.js

```diff
@@ -254,46 +236,45 @@ class Post {
       data.content = content;
 
       // Run "before_post_render" filters
-      return ctx.execFilter('before_post_render', data, {context: ctx});
+      return ctx.execFilter('before_post_render', data, { context: ctx });
     }).then(() => {
+      // Escape content wrapped in <escape></escape>
       data.content = cacheObj.escapeContent(data.content);
 
       if (isSwig) {
-        // Render with Nunjucks if this is a swig file
+        // Render with Nunjucks if the post is a swig file
         return tag.render(data.content, data);
       }
```

Comment:
Pending #2903 ? `isSwig` will be removed in this pull request

Reply:
That doesn't matter. If one of two PRs is merged first, then the other will merge and rebased.